profile agent.TestAgentConnectCALeafCert_aclServiceReadDeny

#### Overview of the Issue

Not 100% sure this is repeatable, need to profile this again. Had one run fail on my laptop on an unrelated test.

#### Reproduction Steps

TODO: Double check the state of this test w/ repro steps

### Operating system and Environment details
macOS 10.14.6

### Log Fragments

```
=== RUN   TestAgentConnectCALeafCert_aclServiceReadDeny
=== PAUSE TestAgentConnectCALeafCert_aclServiceReadDeny
=== CONT  TestAgentConnectCALeafCert_aclServiceReadDeny
--- FAIL: TestAgentConnectCALeafCert_aclServiceReadDeny (1.18s)
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:58.795085 [WARN] The 'acl_datacenter' field is deprecated. Use the 'primary_datacenter' field instead.
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:58.795106 [WARN] bootstrap = true: do not enable unless necessary
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:58.795259 [DEBUG] tlsutil: Update with version 1
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:58.795624 [DEBUG] tlsutil: OutgoingRPCWrapper with version 1
    writer.go:21: 2020/01/17 10:57:59 [INFO]  raft: Initial configuration (index=1): [{Suffrage:Voter ID:f9c4ba39-0d9e-b080-f026-b0d00e9a756f Address:127.0.0.1:33689}]
    writer.go:21: 2020/01/17 10:57:59 [INFO]  raft: Node at 127.0.0.1:33689 [Follower] entering Follower state (Leader: "")
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.093976 [INFO] serf: EventMemberJoin: Node-f9c4ba39-0d9e-b080-f026-b0d00e9a756f.dc1 127.0.0.1
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.123733 [INFO] serf: EventMemberJoin: Node-f9c4ba39-0d9e-b080-f026-b0d00e9a756f 127.0.0.1
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.123850 [INFO] consul: Adding LAN server Node-f9c4ba39-0d9e-b080-f026-b0d00e9a756f (Addr: tcp/127.0.0.1:33689) (DC: dc1)
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.123853 [INFO] consul: Handled member-join event for server "Node-f9c4ba39-0d9e-b080-f026-b0d00e9a756f.dc1" in area "wan"
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.124114 [INFO] agent: Started DNS server 127.0.0.1:33660 (tcp)
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.124136 [INFO] agent: Started DNS server 127.0.0.1:33660 (udp)
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.124482 [INFO] agent: Started HTTP server on 127.0.0.1:33661 (tcp)
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.124505 [INFO] agent: started state syncer
    writer.go:21: 2020/01/17 10:57:59 [WARN]  raft: Heartbeat timeout from "" reached, starting election
    writer.go:21: 2020/01/17 10:57:59 [INFO]  raft: Node at 127.0.0.1:33689 [Candidate] entering Candidate state in term 2
    writer.go:21: 2020/01/17 10:57:59 [INFO]  raft: Election won. Tally: 1
    writer.go:21: 2020/01/17 10:57:59 [INFO]  raft: Node at 127.0.0.1:33689 [Leader] entering Leader state
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.237773 [INFO] consul: cluster leadership acquired
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.237884 [INFO] consul: New leader elected: Node-f9c4ba39-0d9e-b080-f026-b0d00e9a756f
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.279190 [INFO] acl: initializing acls
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.321059 [INFO] acl: initializing acls
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.321126 [INFO] consul: Created ACL 'global-management' policy
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.321145 [WARN] consul: Configuring a non-UUID master token is deprecated
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.363155 [INFO] consul: Created ACL 'global-management' policy
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.363197 [WARN] consul: Configuring a non-UUID master token is deprecated
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.434212 [INFO] consul: Bootstrapped ACL master token from configuration
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.475393 [INFO] consul: Bootstrapped ACL master token from configuration
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.511151 [INFO] consul: Created ACL anonymous token from configuration
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.511231 [INFO] leader: started legacy ACL token upgrade routine
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.511282 [INFO] leader: started acl token reaping routine
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.511315 [DEBUG] acl: transitioning out of legacy ACL mode
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.511585 [INFO] serf: EventMemberUpdate: Node-f9c4ba39-0d9e-b080-f026-b0d00e9a756f
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.511809 [INFO] serf: EventMemberUpdate: Node-f9c4ba39-0d9e-b080-f026-b0d00e9a756f.dc1
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.575160 [INFO] agent: Synced node info
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.575200 [DEBUG] agent: Node info in sync
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.575225 [INFO] consul: Created ACL anonymous token from configuration
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.575330 [INFO] serf: EventMemberUpdate: Node-f9c4ba39-0d9e-b080-f026-b0d00e9a756f
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.575583 [INFO] serf: EventMemberUpdate: Node-f9c4ba39-0d9e-b080-f026-b0d00e9a756f.dc1
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.578774 [DEBUG] consul: dropping node "Node-f9c4ba39-0d9e-b080-f026-b0d00e9a756f" from result due to ACLs
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.704651 [DEBUG] consul CA provider configured ID=07:80:c8:de:f6:41:86:29:8f:9c:b8:17:d6:48:c2:d5:c5:5c:7f:0c:03:f7:cf:97:5a:a7:c1:68:aa:23:ae:81 IsPrimary=true
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.787674 [INFO] agent: Synced service "test-id"
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.787753 [DEBUG] agent: Check "service:test-id" in sync
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.787789 [DEBUG] agent: Node info in sync
    require.go:1159: 
        	Error Trace:	agent_endpoint_test.go:4819
        	Error:      	Should be true
        	Test:       	TestAgentConnectCALeafCert_aclServiceReadDeny
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.854574 [INFO] agent: Requesting shutdown
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.854654 [INFO] consul: shutting down server
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.854682 [DEBUG] leader: stopping legacy ACL token upgrade routine
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.854776 [DEBUG] leader: stopping acl token reaping routine
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.854801 [WARN] serf: Shutdown without a Leave
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.854863 [DEBUG] leader: stopped acl token reaping routine
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.854809 [DEBUG] leader: stopped legacy ACL token upgrade routine
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.908166 [WARN] serf: Shutdown without a Leave
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.915613 [INFO] connect: initialized primary datacenter CA with provider "consul"
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.915718 [INFO] leader: started CA root pruning routine
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.915908 [DEBUG] consul: Skipping self join check for "Node-f9c4ba39-0d9e-b080-f026-b0d00e9a756f" since the cluster is too small
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.916000 [INFO] consul: member 'Node-f9c4ba39-0d9e-b080-f026-b0d00e9a756f' joined, marking health alive
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.928850 [INFO] manager: shutting down
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.950517 [INFO] agent: consul server down
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.950712 [INFO] agent: shutdown complete
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.950881 [INFO] agent: Stopping DNS server 127.0.0.1:33660 (tcp)
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.950672 [ERR] consul: failed to reconcile member: {Node-f9c4ba39-0d9e-b080-f026-b0d00e9a756f 127.0.0.1 33663 map[acls:1 bootstrap:1 build:1.7.0beta2: dc:dc1 id:f9c4ba39-0d9e-b080-f026-b0d00e9a756f port:33689 raft_vsn:3 role:consul segment: vsn:2 vsn_max:3 vsn_min:2 wan_join_port:33664] alive 1 5 2 2 5 4}: leadership lost while committing log
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.951062 [INFO] agent: Stopping DNS server 127.0.0.1:33660 (udp)
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.951328 [INFO] agent: Stopping HTTP server 127.0.0.1:33661 (tcp)
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.951482 [INFO] agent: Waiting for endpoints to shut down
    log.go:172: TestAgentConnectCALeafCert_aclServiceReadDeny - 2020/01/17 10:57:59.951528 [INFO] agent: Endpoints down
FAIL agent.TestAgentConnectCALeafCert_aclServiceReadDeny (1.18s)
```

--- a/agent/agent_endpoint_test.go
+++ b/agent/agent_endpoint_test.go
@@ -4767,6 +4767,7 @@ func TestAgentConnectCALeafCert_aclDefaultDeny(t *testing.T) {
 	a := NewTestAgent(t, t.Name(), TestACLConfig())
 	defer a.Shutdown()
 	testrpc.WaitForLeader(t, a.RPC, "dc1")
+	testrpc.WaitForActiveCARoot(t, a.RPC, "dc1", nil)
 
 	// Register a service with a managed proxy
 	{
@@ -4802,6 +4803,7 @@ func TestAgentConnectCALeafCert_aclServiceWrite(t *testing.T) {
 	a := NewTestAgent(t, t.Name(), TestACLConfig())
 	defer a.Shutdown()
 	testrpc.WaitForLeader(t, a.RPC, "dc1")
+	testrpc.WaitForActiveCARoot(t, a.RPC, "dc1", nil)
 
 	// Register a service with a managed proxy
 	{
@@ -4858,6 +4860,7 @@ func TestAgentConnectCALeafCert_aclServiceReadDeny(t *testing.T) {
 	a := NewTestAgent(t, t.Name(), TestACLConfig())
 	defer a.Shutdown()
 	testrpc.WaitForLeader(t, a.RPC, "dc1")
+	testrpc.WaitForActiveCARoot(t, a.RPC, "dc1", nil)
 
 	// Register a service with a managed proxy
 	{
@@ -4912,6 +4915,7 @@ func TestAgentConnectCALeafCert_good(t *testing.T) {
 	a := NewTestAgent(t, t.Name(), "")
 	defer a.Shutdown()
 	testrpc.WaitForTestAgent(t, a.RPC, "dc1")
+	testrpc.WaitForActiveCARoot(t, a.RPC, "dc1", nil)
 
 	// CA already setup by default by NewTestAgent but force a new one so we can
 	// verify it was signed easily.
@@ -5014,6 +5018,7 @@ func TestAgentConnectCALeafCert_goodNotLocal(t *testing.T) {
 	a := NewTestAgent(t, t.Name(), "")
 	defer a.Shutdown()
 	testrpc.WaitForTestAgent(t, a.RPC, "dc1")
+	testrpc.WaitForActiveCARoot(t, a.RPC, "dc1", nil)
 
 	// CA already setup by default by NewTestAgent but force a new one so we can
 	// verify it was signed easily.

